static/app/views/settings/account/accountIdentities.tsx

```diff
@@ -9,187 +8,208 @@ import {Button} from 'sentry/components/button';
+                <Alert type="error" showIcon>
+                  {tct('Disconnect Your [provider] Identity?', {
+                    provider: identity.provider.name,
+                  })}
+                </Alert>
+                <TextBlock>
+                  {identity.isLogin
+                    ? t(
+                        'After disconnecting, you will need to use a password or another identity to sign in.'
+                      )
+                    : t("This action can't be undone.")}
+                </TextBlock>
+              </Fragment>
+            }
+          >
+            <Button size="sm">{t('Disconnect')}</Button>
+          </Confirm>
+        ) : (
+          <Button
+            size="sm"
+            disabled
+            title={
+              identity.status === UserIdentityStatus.NEEDED_FOR_GLOBAL_AUTH
                 ? t(
-                    'After disconnecting, you will need to use a password or another identity to sign in.'
+                    'You need this identity to sign into your account. If you want to disconnect it, set a password first.'
                   )
-                : t("This action can't be undone.")}
-            </TextBlock>
-          </Fragment>
-        }
-      >
-        <Button size="sm">{t('Disconnect')}</Button>
-      </Confirm>
-    ) : (
-      <Button
-        size="sm"
-        disabled
-        title={
-          identity.status === UserIdentityStatus.NEEDED_FOR_GLOBAL_AUTH
-            ? t(
-                'You need this identity to sign into your account. If you want to disconnect it, set a password first.'
-              )
-            : identity.status === UserIdentityStatus.NEEDED_FOR_ORG_AUTH
-            ? t('You need this identity to access your organization.')
-            : null
-        }
-      >
-        {t('Disconnect')}
-      </Button>
-    );
+                : identity.status === UserIdentityStatus.NEEDED_FOR_ORG_AUTH
+                ? t('You need this identity to access your organization.')
+                : null
+            }
+          >
+            {t('Disconnect')}
+          </Button>
+        )}
+      </InternalContainer>
+    </IdentityPanelItem>
+  );
+}
+
+function AccountIdentities() {
+  const {
+    data: identities = EMPTY_ARRAY,
+    isLoading,
+    isError,
+    refetch,
+  } = useApiQuery<UserIdentityConfig[]>(['/users/me/user-identities/'], {
+    staleTime: 0,
+  });
+
+  const appIdentities = useMemo(
+    () =>
+      identities
+        .filter(identity => identity.category !== UserIdentityCategory.ORG_IDENTITY)
+        .sort(itemOrder),
+    [identities]
+  );
+
+  const orgIdentities = useMemo(
+    () =>
+      identities
+        .filter(identity => identity.category === UserIdentityCategory.ORG_IDENTITY)
+        .sort(itemOrder),
+    [identities]
+  );
+
+  const handleDisconnect = useCallback(
+    (identity: UserIdentityConfig) => {
+      disconnectIdentity(identity, () => refetch());
+    },
+    [refetch]
+  );
```

Comment:
i think refetch won't show the loading state and displays the stale data until the new data appears. We might need to optimistically update or check for isFetching along with isLoading cc @malwilley

Reply:
@ArthurKnaus instead of a refetch you'll want to update the query cache immediately for best UX (so the user knows that something happened). Something like:

```ts
const queryClient = useQueryClient()

disconnectIdentity(identity, () => {
  setApiQueryData(
      queryClient,
      ['/users/me/user-identities/'],
      oldData => {
        if (!Array.isArray(oldData)) {
          return oldData;
        }

        return oldData.filter(i => i.id !== identity.id);
      }
    );
})
```

Reply:
Truly the best option, but not one you need to do (because it would be more work and we just want to convert this), would be to either:

1. Show a loading state while the request is in flight, or
2. Optimistically update the cache immediately (instead of waiting for the request to complete). See this hook for an example of that: https://github.com/getsentry/sentry/blob/229a475155120b567c27e409683fc0c0b91ac959/static/app/views/issueList/mutations/useDeleteSavedSearch.tsx#L33

But once again, no need to improve this page here, just adding some extra context for what we'd do for new components

Reply:
thx for the hints & code snippets! will implement the update of the query cache.
When I find time, I will create a separate PR for the optimistic update + rollback 👍